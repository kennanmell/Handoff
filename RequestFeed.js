/* This will be a list that one can scroll through of requests, and clicking on individual
requests will display a pop up of the requests. The user will also be able to click on the
organization to go to the page of the organization. */

/*
        Things I need to do:
        1. Get a function call to get real data, which needs to be tranformed into requests
        2. Turn those requests into requestcomponents and put them in the bottom thing
        3. Figure out how to send variables through a callback
        4. make a compentent that i can both initialize and pass in props so that we dont have
        to hardcode the data for requests
        */
import React, { Component } from 'react';
import { Alert, Button, TouchableHighlight, TextInput,
            StyleSheet, AppRegistry, ListView, Text, View } from 'react-native';

const styles = StyleSheet.create({
  separator: {
    flex: 1,
    height: 6,
    backgroundColor: '#C1C1C1',
  },
  container: {
      flex: 1,
      padding: 8,
      height: 70,
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: '#C1C1C1',
  },
  input: {
      height: 50,
      flex: 1,
      paddingHorizontal: 8,
      fontSize: 25,
      backgroundColor: '#FFFFFF',
      borderRadius: 2,
  }
});

const Header = (props) => (
  <View style={styles.container}>
    <TextInput
      style={styles.input}
      placeholder="Search..."
      onChangeText={(text) => console.log('searching for ', text)}
    />
  </View>
);

/*function Request(props) {
  return <h1>Hello, {props.name}</h1>;
};
const elem1 = <Request name="Help"/>;
const elem2 = <Request name="Help us"/>;
*/
/*
class RequestComp extends Component {
  constructor(props) {
    super(props);
    //this.request = new Request("Hell yeah", "descrip", "org", "Hell Yeah", ["stuff", "h"]);
    //this.state = {text: ''};
  }

  render() {
    return (
      <View style={{padding: 10}}>
      <Text> Title: _______ Org: ________</Text>
      <Text> Text for the request should go here </Text>
      <Button
        onPress={onPressLearnMore}
        title="Learn More"
        color="#841584"
      />
      </View>
    );
  }
}


const elem1 = new RequestComp();
const elem2 = new RequestComp();*/
var Request = React.createClass ({
    render: function() {
        return (
            <View>
                <Text style={{fontSize: 20}}>  Title: ________</Text>
                <Text style={{fontSize: 20}}>  Org: ________</Text>
                <Text style={{alignItems: 'center'}, {fontSize: 18}}> Text for the request should
                    go here. It can talk about how we needs some donations or something just want
                    so get the format in here.
                </Text>
            </View>
        );
    }

    /* more info button that looks terrible
                <TouchableHighlight
                    //onPress= {Alert.alert('This should display the info of the organization')}
                    underlayColor={'#612e5f'}>
                    <Text style={{alignSelf: 'flex-end'},{fontSize: 15}}>More Info</Text>
                </TouchableHighlight>*/
});

export default class RequestFeed extends Component {
    constructor(props) {
        super(props);
        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.state = {

          dataSource: ds.cloneWithRows([
            <Request/>, <Request/>, <Request/>, <Request/>,
            <Request/>, <Request/>, <Request/>, <Request/>,
            <Request/>, <Request/>, <Request/>, <Request/>
          ])
        };
    }

    render() {
      return (
        <View>
          <ListView
            dataSource={this.state.dataSource}
            renderRow={(rowData) => <Request>{rowData}</Request>}
            renderSeparator={(sectionId, rowId) => <View key={rowId} style={styles.separator} />}
            renderHeader={() => <Header />}
          />
        </View>
      );
    }
}
