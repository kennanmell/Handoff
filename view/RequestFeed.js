/* This will be a list that one can scroll through of requests, and clicking on individual
requests will display a pop up of the requests. The user will also be able to click on the
organization to go to the page of the organization. */

/*
        Things I need to do:
        1. Get a function call to get real data, which needs to be tranformed into requests
        2. Turn those requests into requestcomponents and put them in the bottom thing
        3. Figure out how to send variables through a callback
        4. make a compentent that i can both initialize and pass in props so that we dont have
        to hardcode the data for requests
        */
import React, { Component, PropTypes } from 'react';
import { TouchableHighlight, TextInput,
            StyleSheet, AppRegistry, ListView, Text, View } from 'react-native';
//import Request from './Request';

// These are different styles
const styles = StyleSheet.create({
  separator: {
    flex: 1,
    height: 6,
    backgroundColor: '#C1C1C1',
  },
  container: {
      flex: 1,
      padding: 8,
      height: 70,
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: '#C1C1C1',
  },
  input: {
      height: 50,
      flex: 1,
      paddingHorizontal: 8,
      fontSize: 25,
      backgroundColor: '#FFFFFF',
      borderRadius: 2,
  }
});

// This header is a search bar and it actually looks good.
const Header = (props) => (
  <View style={styles.container}>
    <TextInput
      style={styles.input}
      placeholder="Search..."
      onChangeText={(text) => console.log('searching for ', text)}
    />
  </View>
);

// Returns a promise
async function getRequests() {
    return fetch('https://u116vqy0l2.execute-api.us-west-2.amazonaws.com/prod/requests', {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
          limit: 30,
      })
    })
}

const Row = (props) => (
  <View style={{padding: 10}}>
    <Text style={{fontSize: 20}}>  {props.title}</Text>
    <Text style={{fontSize: 20}}>  {props.organization}</Text>
    <Text style={{alignItems: 'center'}, {fontSize: 18}}>  {props.description}
    </Text>
  </View>
);

export default class RequestFeed extends Component {
    constructor(props) {
        super(props);
        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        requests = [{
           "organization": "HandOff Team",
           "time": 7,
           "title": "Waiting on data",
           "description": "Should be fetching requests, if this takes too long, check" +
               "your internet connection.",
        }];

        getRequests()
            .then((response) => response.json())
            .then((responseJson) => {
                //console.log('title below');
                //console.log(responseJson.requests[0].title);
                //result = [responseJson.requests[0], responseJson.requests[1], responseJson.requests[3]];

                this.setState({
                  dataSource: ds.cloneWithRows(responseJson.requests)
                });
              })
              .catch((error) => {
                console.error(error);
        });

       this.state = {
            dataSource: ds.cloneWithRows(requests)
        }

    }

    render() {
      return (
        <View>
          <ListView
            dataSource={this.state.dataSource}
            renderRow={(rowData) => <Row {...rowData} />}
            renderSeparator={(sectionId, rowId) => <View key={rowId} style={styles.separator} />}
            renderHeader={() => <Header />}
          />
        </View>
      );
    }
}
