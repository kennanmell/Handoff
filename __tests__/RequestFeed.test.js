import 'react-native';
import React from 'react';
import RequestFeed from '../View/RequestFeed';
import renderer from 'react-test-renderer';

// Ensures that the donator request feed page (RequestFeed) renders correctly.
it('renders correctly', () => {
  const tree = renderer.create(
    	<RequestFeed />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});