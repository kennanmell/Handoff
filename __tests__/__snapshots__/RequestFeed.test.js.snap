exports[`test renders correctly 1`] = `
<View>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderHeader={[Function]}
    renderRow={[Function]}
    renderSeparator={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1C1C1",
          "flex": 1,
          "flexDirection": "row",
          "height": 70,
          "padding": 8,
        }
      }>
      <TextInput
        onChangeText={[Function]}
        placeholder="Search..."
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 2,
            "flex": 1,
            "fontSize": 25,
            "height": 50,
            "paddingHorizontal": 8,
          }
        } />
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }>
          
        Waiting on data
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }>
          
        HandOff Team
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }>
          
        Should be fetching requests, if this takes too long, checkyour internet connection.
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#C1C1C1",
          "flex": 1,
          "height": 6,
        }
      } />
  </ScrollView>
</View>
`;
